### DISCUSSION TITLE
Remove dependency on `EnumerableAsyncProcessor`

### DISCUSSION THREAD

**Discussion by robertcoltheart** [thomhurst/TUnit#1455] [Ideas]
Hear me out! 😄 I'm not opposed to TUnit using it for it's running, since it clearly leverages this library for parallel processing etc. But as a test framework, I really only want the bare minimum of dependencies pulled into my tests, and this doesn't feel like something I want forced into my dependency tree.

What about source-compiling this into TUnit as internal classes instead?

**Comment by thomhurst** [comment]
I'd have the maintain two copies of the code then if I need to make any changes

**Reply by robertcoltheart** [reply]
There are ways around this. The easiest option would be using a git submodule, where you are just pulling in the code using git and compiling it. Another option would be to publish a source-only package and then consume that package here.

**Reply by thomhurst** [reply]
Hmm... Is the dependency causing you issues?

**Reply by robertcoltheart** [reply]
I guess from a design standpoint, all the major testing frameworks only have dependencies on their own packages. But from a coding perspective, I guess I just don't want to have a bunch of extension methods polluting my intellisense that I'm likely not going to use. I have the same issue with libraries that expose extension methods that should really be internal. Castle.Core is a glaring example of this, and my team will occasionally use one of its extension methods that is in an "internal" namespace.

**Reply by robertcoltheart** [reply]
For what it's worth, I'm being a purist and this is quite nit-picking 😊

**Reply by thomhurst** [reply]
I'll see if I can unpick it, but it won't be my highest priority for now :)

**Reply by robertcoltheart** [reply]
I'm happy to contribute if there is a particular direction you'd like to take.

Thanks for injecting some life into the unit test scene! ❤️
